iabtcf-encoder/src/main/java/com/iabtcf/encoder/VendorFieldEncoder.java

```diff
@@ -0,0 +1,227 @@
+        if (vendorId <= 0) {
+            throw new IndexOutOfBoundsException("vendorId < 1: " + vendorId);
+        }
+        vendors.set(vendorId - 1);
+
+        return this;
+    }
+
+    public VendorFieldEncoder add(Iterable<Integer> vendorIds) {
+        for (Iterator<Integer> i = vendorIds.iterator(); i.hasNext();) {
+            add(i.next());
+        }
+
+        return this;
+    }
+
+    public VendorFieldEncoder add(int... vendorIds) {
+        for (int i = 0; i < vendorIds.length; i++) {
+            add(vendorIds[i]);
+        }
+
+        return this;
+    }
+
+    public VendorFieldEncoder add(IntIterable ii) {
+        for (IntIterator i = ii.intIterator(); i.hasNext();) {
+            add(i.nextInt());
+        }
+
+        return this;
+    }
+
+    /**
+     * Emit the specified max vendor id. By default, the maximum vendorId that was added is emitted.
+     *
+     * @VisibleForTesting
+     */
+    VendorFieldEncoder setMaxVendorId(int maxVendorId) {
+        this.maxVendorId = maxVendorId;
+
+        return this;
+    }
+
+    public BitWriter buildV1() {
+        return build(true);
+    }
+
+    /**
+     * Returns a BitWriter that represents the encoded vendor ids. The BitWriter may either encode a
+     * bit field or a range encoding; depending on which is smaller.
+     */
+    public BitWriter build() {
+        return build(false);
+    }
+
+    private BitWriter build(boolean emitDefaultConsent) {
+        BitWriter bv = new BitWriter();
+
+        if (vendors.length() == 0) {
+            bv.writeV(0, FieldDefs.CORE_VENDOR_MAX_VENDOR_ID);
+            bv.write(false, FieldDefs.CORE_VENDOR_IS_RANGE_ENCODING);
+            return bv;
+        }
+
+        maxVendorId = Math.max(vendors.length(), maxVendorId);
+
+        // create the range bit section
+        BitWriter rangeBits = new BitWriter();
+        int idxSet = vendors.get(0) ? 0 : vendors.nextSetBit(0);
+        int idxClr;
+        int numEntries = 0;
+        do {
+            idxClr = vendors.nextClearBit(idxSet);
+            int length = idxClr - idxSet;
+
+            if (length == 1) {
+                rangeBits.write(false, FieldDefs.CORE_VENDOR_IS_RANGE_ENCODING);
+                rangeBits.writeV(idxSet + 1, FieldDefs.START_OR_ONLY_VENDOR_ID);
+            } else {
+                rangeBits.write(true, FieldDefs.CORE_VENDOR_IS_RANGE_ENCODING);
+                rangeBits.writeV(idxSet + 1, FieldDefs.START_OR_ONLY_VENDOR_ID);
+                rangeBits.writeV(idxClr, FieldDefs.END_VENDOR_ID);
+            }
+            numEntries++;
+        } while ((idxSet = vendors.nextSetBit(idxClr)) > 0
+                && (rangeBits.length() < vendors.length() || emitRangeEncoding));
+
+        // emit max vendor id
+        if (emitMaxVendorId) {
+            bv.writeV(maxVendorId, FieldDefs.CORE_VENDOR_MAX_VENDOR_ID);
+        }
+
+        if (rangeBits.length() < vendors.length() || emitRangeEncoding) {
```

Comment:
extract for better readability

Reply:
what's the readability issue here?

Reply:
it just reads better when you extract such line into separate method with meaningful name

`if(isEligibleToPerform()) { ....`

Reply:
i see, i'd have to disagree